src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java

```diff
@@ -125,71 +95,137 @@ public JPanel getOptionsPanel() {
 
         try {
-            URL url = new URL(searchUrl);
-            String page = Util.getResults(url);
+            URL url = new URL(IEEEXploreFetcher.URL_SEARCH);
 
-            if (page.contains("You have entered an invalid search")) {
-                status.showMessage(Localization.lang("You have entered an invalid search '%0'.", terms),
-                        DIALOG_TITLE, JOptionPane.INFORMATION_MESSAGE);
-                return false;
-            }
+            HttpURLConnection con = (HttpURLConnection) url.openConnection();
+
+            //add request header
+            con.setRequestMethod("POST");
+            con.setRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 5.1; rv:31.0) Gecko/20100101 Firefox/31.0");
+            con.setRequestProperty("Accept", "application/json");
+            con.setRequestProperty("Content-type", "application/json");
 
-            if (page.contains("Bad request")) {
-                status.showMessage(Localization.lang("Bad Request '%0'.", terms),
+            String page = Util.getPostResultsWithEncoding(con, postData, null);
+            //the page can be blank if the search did not work (not sure the exact conditions that lead to this, but declaring it an invalid search for now)
+            if (page.isEmpty()) {
+                status.showMessage(Localization.lang("You have entered an invalid search '%0'.", query),
                         DIALOG_TITLE, JOptionPane.INFORMATION_MESSAGE);
                 return false;
             }
 
-            if (page.contains("No results were found.")) {
-                status.showMessage(Localization.lang("No entries found for the search string '%0'", terms),
+            //parses the JSON data returned by the query
+            //todo: a faster way would be to parse the JSON tokens one at a time just to extract the article number, but this seems to be fast enough...
+            JSONObject jsonObj = new JSONObject(page);
+            hits = jsonObj.getInt("totalRecords");
+
+            //if no search results were found
+            if (hits == 0) {
+                status.showMessage(Localization.lang("No entries found for the search string '%0'", query),
                         DIALOG_TITLE, JOptionPane.INFORMATION_MESSAGE);
                 return false;
             }
 
-            if (page.contains("Error Page")) {
+            //if max hits were exceeded, display the warning
+            if (hits > IEEEXploreFetcher.MAX_FETCH) {
                 // @formatter:off
-                status.showMessage(
-                        Localization.lang("Intermittent errors on the IEEE Xplore server. Please try again in a while."),
-                       DIALOG_TITLE, JOptionPane.INFORMATION_MESSAGE);
+                status.showMessage(Localization.lang("%0 entries found. To reduce server load, only %1 will be downloaded.",
+                        new String[] {String.valueOf(hits), String.valueOf(IEEEXploreFetcher.MAX_FETCH)}),
```

Comment:
You do not need to pass a String array, just present the two parameters as additional parameters to the lang method.